frontend/src/v5/ui/routes/dashboard/projects/tickets/ticketsTable/ticketsTableContent/ticketsTableContent.component.tsx

```diff
@@ -16,66 +16,40 @@
+import { ITemplate } from '@/v5/store/tickets/tickets.types';
+import { TicketsTableContextComponent } from '../ticketsTableContext/ticketsTableContext';
+import { getAvailableColumnsForTemplate } from '../ticketsTableContext/ticketsTableContext.helpers';
+import { BaseProperties } from '@/v5/ui/routes/viewer/tickets/tickets.constants';
 
-export const TicketsTableContent = (props: TicketsTableResizableContentProps) => {
+const TableContent = ({ template, ...props }: TicketsTableResizableContentProps & { template: ITemplate }) => {
 	const { filteredItems } = useContext(SearchContext);
-	const { template: templateId } = useParams<DashboardTicketsParams>();
-	const [modelsIds] = useSearchParam('models', Transformers.STRING_ARRAY);
-
-	const template = ProjectsHooksSelectors.selectCurrentProjectTemplateById(templateId);
-	const { config, modules } = template;
-	const hasProperties = config?.issueProperties;
-	const hasSafetibase = modules?.some((module) => module.type === 'safetibase');
-	const showModelName = modelsIds.length > 1;
+	const { stretchTable, getVisibleColumnsNames, showColumn } = useContext(ResizableTableContext);
+	const templateWasFetched = templateAlreadyFetched(template);
+	const hasVisibleColumns = getVisibleColumnsNames().length > 0;
 	
-	const columns: TableColumn[] = [
-		{ name: 'id', width: 80, minWidth: 25 },
-		{ name: BaseProperties.TITLE, width: 380, minWidth: 25, stretch: true },
-		{ name: 'modelName', width: 145, minWidth: 25 },
-		{ name: `properties.${BaseProperties.CREATED_AT}`, width: 127, minWidth: 25 },
-		{ name: `properties.${IssueProperties.ASSIGNEES}`, width: 96, minWidth: 25 }, 
-		{ name: `properties.${BaseProperties.OWNER}`, width: 52, minWidth: 25 },
-		{ name: `properties.${IssueProperties.DUE_DATE}`, width: 147, minWidth: 25 },
-		{ name: `properties.${IssueProperties.PRIORITY}`, width: 90, minWidth: 25 },
-		{ name: `properties.${BaseProperties.STATUS}`, width: 150, minWidth: 52 },
-		{ name: `modules.safetibase.${SafetibaseProperties.LEVEL_OF_RISK}`, width: 137, minWidth: 25 },
-		{ name: `modules.safetibase.${SafetibaseProperties.TREATMENT_STATUS}`, width: 134, minWidth: 25 },
-	];
-
-	const getHiddenColumns = () => {
-		const cols = [];
-		if (!showModelName) {
-			cols.push('modelName');
-		}
-		if (!hasProperties) {
-			cols.push(
-				`properties.${IssueProperties.ASSIGNEES}`,
-				`properties.${IssueProperties.DUE_DATE}`,
-				`properties.${IssueProperties.PRIORITY}`,
-			);
+	useEffect(() => {
+		if (templateWasFetched && hasVisibleColumns) {
+			stretchTable([BaseProperties.TITLE]);
 		}
-		if (!hasSafetibase) {
-			cols.push(
-				`modules.safetibase.${SafetibaseProperties.LEVEL_OF_RISK}`,
-				`modules.safetibase.${SafetibaseProperties.TREATMENT_STATUS}`,
-			);
+	}, [template, templateWasFetched, hasVisibleColumns]);
+
+	useEffect(() => {
+		if (templateWasFetched && !hasVisibleColumns) {
+			showColumn('id');
 		}
-		return cols;
-	};
+	}, [hasVisibleColumns, templateWasFetched]);
```

Comment:
what if get rid of this management of which column is visible and just dont let users get rid of every column in the cog?

edit: I saw in the cog code that you are already managing that, then why is this here?